command/git.go

```diff
@@ -152,6 +154,9 @@ func (opts cloneOpts) buildHTTPCloneCommand() ([]string, error) {
 		return nil, errors.Wrap(err, "failed to parse URL from location")
 	}
 	clone := fmt.Sprintf("git clone %s '%s'", thirdparty.FormGitUrl(urlLocation.Host, opts.owner, opts.repo, opts.token), opts.dir)
+	if !opts.mergeTestRequester && opts.shallowClone {
```

Comment:
why not shallow clone for commit queue tasks as well?

Reply:
(as long as the potential `git fetch --unshallow` would happen before we try to check out the branch)

Reply:
According to the man page, `--depth` implies `--single-branch`. I was deciding among the following:
- One option would be to pass `--no-single-branch`. However, this will "fetch the histories near the tips of all branches." Since some repos in Evergreen do a pull request repo against upstream instead of a fork, this could potentially be a large number of branches.
- Another option would be to undo `--single-branch`. This is covered in this StackOverflow answer https://stackoverflow.com/a/17937889. It requires a deep enough knowledge of git internals (you can't learn this directly from the `git clone` or `git fetch` man pages) that I hesitate to use it in our code base.
- The third option is to exclude the commit queue altogether. Since commit queue patches are far less frequent than other patches, we save much less shallow cloning with them.

Ultimately we could potentially shallow clone, but I was concerned about the performance of the first option and the complexity of the second. I may not be reading this carefully enough, but I decided to de-risk this by not including merge patches. I think it'd be fine to do future work to speed up merge patches as well.

Reply:
could we pass the project's branch to `--single-branch`?

Reply:
We could, but we `git checkout`. I'm pushing to keep the intervention here pretty narrow.

Reply:
Talked in person. Going to test a slightly more substantial change in staging.